cmd/d2/watch.go

```diff
@@ -325,7 +329,15 @@ func (w *watcher) compileLoop(ctx context.Context) error {
 			recompiledPrefix = "re"
 		}
 
-		b, err := compile(ctx, w.ms, w.layoutPlugin, w.inputPath, w.outputPath)
+		if filepath.Ext(w.outputPath) == ".png" && !w.pw.Browser.IsConnected() {
+			newPW, err := w.pw.RestartBrowser()
+			if err != nil {
```

Comment:
u still want to broadcast the error. during `watch`, the user might have their browser and editor open (no terminal), and it'd just look frozen, so you want to give a human-friendly error msg broadcast like "issue encountered with PNG exporter, stopping".

btw everywhere else here doesn't return because it's recoverable (they just change the D2 text to be well-formed), but here it's not recoverable.